cloudinit/sources/DataSourceOracle.py

```diff
@@ -192,13 +127,20 @@ def _get_data(self):
 
         # network may be configured if iscsi root.  If that is the case
         # then read_initramfs_config will return non-None.
-        if _is_iscsi_root():
-            data = read_opc_metadata()
-        else:
-            with dhcp.EphemeralDHCPv4(net.find_fallback_nic()):
-                data = read_opc_metadata()
+        fetch_vnics_data = self.ds_cfg.get('configure_secondary_nics', False)
```

Comment:
I'm not sure I understand what you mean by 'defaulting of `configure_secondary_nics`'

Reply:
This line determines what the default value of `configure_secondary_nics` will be if it isn't present in `self.ds_cfg`: `False`. In `.network_config`, we also determine what the default value of `configure_secondary_nics` will be if it isn't present (and in that case as we don't supply a second argument to `.get`, it will be `None`).

Though, further confusing matters, we actually set the default in `BUILTIN_DS_CONFIG` which _should_ mean that `configure_secondary_nics` is always set in `self.ds_cfg` (so we could use regular dict access instead of `.get`). However, that goes through `util.mergemanydict` and I don't know for sure that that couldn't result in a dict without the key present under certain user inputs.

Does that make any more sense?

Reply:
Yes, mainly because of this
> so we could use regular dict access instead of .get

I guess I don't see the problem with using `.get` here (or anywhere and everything :wink: ). Since we're not persisting the result of this anywhere, I don't see it as "defaulting a value", but rather eliminating the need for None checks in case something hasn't been set. Why would having things as they are increase the maintenance burden?